plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/meta/CubridMetaModel.java

```diff
@@ -53,100 +65,362 @@ public String getTableOrViewName(GenericTableBase base)
             @NotNull JDBCSession session,
             @NotNull GenericStructContainer owner,
             @Nullable GenericTableBase forTable)
-            throws SQLException
-    {
-        return session.getMetaData().getColumns(null, null, this.getTableOrViewName(forTable), null).getSourceStatement();
+            throws SQLException {
+        String sql = "show full columns from " + getTableOrViewName(forTable);
+        final JDBCPreparedStatement dbStat = session.prepareStatement(sql);
+        return dbStat;
     }
 
+    @NotNull
     @Override
     public JDBCStatement prepareUniqueConstraintsLoadStatement(
             @NotNull JDBCSession session,
             @NotNull GenericStructContainer owner,
             @Nullable GenericTableBase forTable)
-            throws SQLException, DBException
-    {
-        return session.getMetaData().getPrimaryKeys(null, null, this.getTableOrViewName(forTable)).getSourceStatement();
+            throws SQLException, DBException {
+        CubridTable table = (CubridTable) forTable;
+        String sql = "select *, t1.index_name as PK_NAME from db_index t1 join db_index_key t2 \n"
+                + "on t1.index_name = t2.index_name where is_unique = 'YES' and t1.class_name = ? \n"
+                + (table.getDataSource().getSupportMultiSchema() ? "and t1.owner_name = ?" : "");
+        final JDBCPreparedStatement dbStat = session.prepareStatement(sql);
+        dbStat.setString(1, table.getName());
+        if(table.getDataSource().getSupportMultiSchema()) {
+            dbStat.setString(2, table.getSchema().getName());
+        }
+        return dbStat;
+    }
+
+    @Nullable
+    @Override
+    public GenericTableConstraintColumn[] createConstraintColumnsImpl(
+            @NotNull JDBCSession session,
+            @NotNull GenericTableBase parent,
+            @Nullable GenericUniqueKey object,
+            @Nullable GenericMetaObject pkObject,
+            @NotNull JDBCResultSet dbResult)
+            throws DBException {
+        String name = JDBCUtils.safeGetStringTrimmed(dbResult, "key_attr_name");
+        Integer keyOrder = JDBCUtils.safeGetInteger(dbResult, "key_order") + 1;
+        GenericTableColumn tableColumn = parent.getAttribute(session.getProgressMonitor(), name);
+        return new GenericTableConstraintColumn[] {
+              new GenericTableConstraintColumn(object, tableColumn, keyOrder) };
     }
 
+    @Nullable
+    @Override
+    public DBSEntityConstraintType getUniqueConstraintType(@NotNull JDBCResultSet dbResult) throws DBException {
+        String isUnique = JDBCUtils.safeGetString(dbResult, "is_unique");
+        String isPrimary = JDBCUtils.safeGetString(dbResult, "is_primary_key");
+        if(isPrimary.equals("YES")) {
+            return DBSEntityConstraintType.PRIMARY_KEY;
+        } else if (isUnique.equals("YES")) {
+            return DBSEntityConstraintType.UNIQUE_KEY;
+        } else {
+            return null;

```

Comment:
Why null? What type of object is it then? Why not the PK by default?

Reply:
We will modify the condition here. And FYI, Cubrid supports unique key. So, if it is not primary key, then it is unique key by default.

Reply:
I have fixed this.